src/metabase/email.clj

```diff
@@ -129,6 +130,70 @@
       (log/warn e (trs "Failed to send email"))
       {::error e})))
 
+(defn- reconfigure-retrying "No-op. Will be overridden." [_ _])
```

Comment:
should this just be `(declare reconfigure-retrying)`? Or is the private metadata messing with it?

Reply:
The problem is, this function is getting called before the real definition is read. It seems when the settings get their initial values, the update function gets called. (Which I think is good thing.)

Reply:
Unfortunately, the current "solution" is not just ugly, it triggers the linter too.

Reply:
Are you sure? I tried replacing this with `declare` locally and everything seems to work as expected.